TikTokApi/browser.py

```diff
@@ -111,6 +114,47 @@ def create_page(self, set_useragent=False):
 
         return page
 
+    def base36encode(self, number, alphabet='0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ'):
+        """Converts an integer to a base36 string."""
+        base36 = ''
+        sign = ''
+
+        if number < 0:
+            sign = '-'
+            number = -number
+
+        if 0 <= number < len(alphabet):
+            return sign + alphabet[number]
+
+        while number != 0:
+            number, i = divmod(number, len(alphabet))
+            base36 = alphabet[i] + base36
+
+        return sign + base36
+
+    def gen_verifyFp(self):
+        start_time = int(time.time() * 1000)
+        chars = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"[:]
+        chars_len = len(chars)
+        scenarioTitle = self.base36encode(int(time.time() * 1000))
+        uuid = [0] * 36
+        uuid[8] = '_'
+        uuid[13] = '_'
+        uuid[18] = '_'
+        uuid[23] = '_'
+        uuid[14] = "4"
+        r = None
+        for i in range(36):
+            if uuid[i] == 0:
+                if r == None:
+                    r = 0
+                else:
+                    r = random.random() * chars_len
+                uuid[i] = chars[int(r)]
+        ending = ""
+        for x in uuid:
+            ending += str(x)
+        return "verify_" + scenarioTitle + "_" + ending
```

Comment:
This is much more complex than the generator function I wrote, but mine's completely random, just matching the pattern roughly.

How did you come up with this algorithm? It seems like you understand the structure of these tokens much more than I do.

Reply:
In the captcha javascript file which you can find a link to above there's a function that creates them in that file. I just converted the JS to python

Reply:
Ah, I see! well done! I see a few small tweaks that could be made to make it more pythonic. I'll try to make a PR for those.  The most important one is to use the identity operator `is` when comparing to `None`.  `None` is always a singleton in python, so the identity operator is more appropriate for detecting it.  I believe that also comes with a slight performance benefit as well.

Reply:
Hi! In the origin function ( captcha.js -> glkote_init ): `uuid[i] = chars[19 == i ? 3 & r | 8 : r];`
But in the your code: `uuid[i] = chars[int(r)]`
Why? 

_I am from Russia and the translation may not be correct_

Reply:
And in the origin code ( captcha.js ) function return scenarioTitle in lower case, but your code generate scenarioTitle in upper case. 
I think the more correct code is below
`def glkote_init() -> str:
	chars = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"[:]
	chars_len = len(chars)
	scenarioTitle = base36_encode(int(time.time() * 1000))
	uuid = [0] * 36
	uuid[8] = uuid[13] = uuid[18] = uuid[23] = '_'
	uuid[14] = '4'

	for i in range(36):
		if uuid[i] != 0:
			continue
		r = int(random.random() * chars_len)
		uuid[i] = chars[int((3 & r) | 8 if 19 == i else r)]

	return f'verify_{scenarioTitle.lower()}_{"".join(uuid)}'`